Examples/Detectors/Geant4Detector/src/MockupSectorBuilder.cpp

```diff
@@ -0,0 +1,344 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022-2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/Geant4Detector/MockupSectorBuilder.hpp"
+
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Detector/PortalGenerators.hpp"
+#include "Acts/Detector/PortalHelper.hpp"
+#include "Acts/Geometry/CuboidVolumeBounds.hpp"
+#include "Acts/Geometry/CylinderVolumeBounds.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Navigation/SurfaceCandidatesUpdators.hpp"
+#include "Acts/Plugins/Geant4/Geant4Converters.hpp"
+#include "Acts/Plugins/Geant4/Geant4DetectorSurfaceFactory.hpp"
+#include "Acts/Plugins/Geant4/Geant4PhysicalVolumeSelectors.hpp"
+#include "Acts/Surfaces/StrawSurface.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "ActsExamples/Geant4/GdmlDetectorConstruction.hpp"
+#include "ActsExamples/Geant4Detector/Geant4Detector.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+
+// a function for sorting the detector volumes from the innermost to the
+// outermost
+
+bool sortByRadialDistance(
+    std::shared_ptr<Acts::Experimental::DetectorVolume>& det_vol1,
+    std::shared_ptr<Acts::Experimental::DetectorVolume>& det_vol2) {
+  return det_vol1->center().y() < det_vol2->center().y();
+}
```

Comment:
I think this is simple enough to go into a lambda

otherwise it should go into an anonymous namespace